tools/lib/NQP/Config/Rakudo.pm

```diff
@@ -652,7 +652,8 @@ sub gen_nqp {
 
         unless ( $force_rebuild
             || !$nqp_have
-            || $nqp_ver_ok )
+            || $nqp_ok
+            || defined($gen_nqp))
```

Comment:
Wouldn't it force rebuild nqp _always_ if `--gen-nqp` is used? If so, I disagree. What's the point if NQP version satisfies us? `--force-rebuild` is there to unconditionally do the rebuild if needed.

Reply:
I don't think that's what happens. This logic is only about the version mismatch check. The idea being: If we passed `--gen-nqp` don't fail with an "is outdated" error should the version be stale.
A few lines later another `if` determines whether we need to rebuild and that logic does indeed not rebuild if the version is fine.